src/directives/infinite-scroll/infinite-scroll.component.spec.ts

```diff
@@ -0,0 +1,127 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { InfiniteScrollModule } from './infinite-scroll.module';
+import { Component } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+
+
+@Component({
+    template: `<div class="row">
+                    <div class="col-md-6 col-sm-12">
+                        <input id="filter" type="text" class="form-control" placeholder="Filter"
+                            [(ngModel)]="filterText">
+                    </div>
+                </div>
+                <div class="row m-t-sm">
+                    <div class="col-md-6 col-sm-12">
+
+                        <div class="employee-list"
+                            [uxInfiniteScroll]="load"
+                            [filter]="filterText"
+                            [pageSize]="pageSize"
+                            [loadOnScroll]="loadOnScroll">
+
+                        </div>
+
+                    </div>
+                </div>
+    `
+})
+export class InfiniteScrollTestComponent {
+
+    filterText: any;
+    pageSize = 20;
+    loadOnScroll: boolean = false;
+    loading: boolean = false;
+    exhausted: boolean = false;
```

Comment:
I think the only property we need here is `filterText`. `loading` and `exhausted` are never used and `pageSize` and `loadOnScroll` never change so can just be put in the html, eg: `[loadOnScroll]="true"`